src/components/password.components.js

```diff
@@ -0,0 +1,89 @@
+import $ from 'jquery';
+
+export default class PasswordInput {
+    constructor(queryString, valid = false) {
+        this.div = $(queryString);
+        this.input = $(`${queryString} input[type="password"]`);
+        this.icon = $(`${queryString} input[type="password"]+i`);
+
+        this.valid = valid;
+
+        this.checkQuery(this.div, this.input, this.icon);
+        this.onClick();
+    }
+
+    checkQuery(div, input, icon) {
+        if (div.length === 0) {
+            throw new Error('Failed to query div');
+        }
+
+        if (input.length === 0) {
+            throw new Error('Failed to query input password');
+        }
+
+        if (icon.length === 0) {
+            throw new Error('Failed to query icon');
+        }
+    }
+
+    onClick() {
+        this.icon.on('click', () => {
+            if (this.input.attr('type') === 'password') {
+                this.input.attr('type', 'text').trigger('focus');
+                this.icon.removeClass('fa-eye-slash').addClass('fa-eye');
+            } else {
+                this.input.attr('type', 'password').trigger('focus');
+                this.icon.removeClass('fa-eye').addClass('fa-eye-slash');
+            }
+        });
+    }
+
+    getValue() {
+        return this.input.val();
+    }
+
+    setValue(string) {
+        this.input.val(string);
+    }
+
+    isValid() {
```

Comment:
You guys must use the same name for the validation method. And must have the same signature and return.